labs/idp-okta-integration/README.md

```diff
@@ -0,0 +1,273 @@
+- In a real-world scenario, the redirection would be back to the client application and it would parse Okta's response to capture the authorization code
+
+![Okta auth code](assets/okta-auth-code-response.png)
+
+4. Pass the authorization code, client id, and base64 encoded string to Apigee Identity Facade to generate a Bearer token that will be used for API calls.
+```
+export AUTH_CODE={authorization code returned above}
+export APIGEE_RESPONSE=$(curl -s --location --request POST "https://$APIGEE_X_HOSTNAME/v1/oauth20/token?client_id=$APIGEE_CLIENT_ID" \
+--header "Authorization: Basic $BASE64_ENCODED" \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--data-urlencode 'redirect_uri=https://httpbin.org/get' \
+--data-urlencode 'grant_type=authorization_code' \
+--data-urlencode "code=$AUTH_CODE")
+echo $APIGEE_RESPONSE
+export ACCESS_TOKEN=$(echo $APIGEE_RESPONSE | jq -r .access_token)
+echo $ACCESS_TOKEN
+```
+5. Apigee will respond with a Bearer token (access_token) which is saved to an environment variable
+
+
+6. Finally, call an endpoint protected by OAuth. The identity facade has /protected one configured to validate OAuth tokens.
+
+```
+curl --location --request GET "https://$APIGEE_X_HOSTNAME/v1/oauth20/protected" \
+--header "Authorization: Bearer $ACCESS_TOKEN"
+```
+![IDP Test](assets/idp-test-proxy.png)
+
+## Use Identity Facade with API Proxy
+
+The default Apigee install includes a /hello-world proxy. In this section we will add an OAuthv2 policy that verifies the access token and create an application that has access to the identity facade (generate tokens) and the Hellow-World proxies.
+
+1. In the Apigee portal, expand Publish and click on API Products.
+2. Choose + CREATE
+3. Fill in the following
+
+- Name: Hellow World
+- Display Name: Hello World
+- Environment: eval
+- Access: Public
+- Operations (ADD AN OPERATION)
+     - API Proxy: hello-world
+     - Path: /
+     - Save
+- Scroll up and click Save
+
+![API Product1](assets/api-product1.png)
+![API Product2](assets/api-product2.png)    
+
+3. Under Public, click on Apps
+- Notice there is an app created called identity app. The client id and secret used above are from this application. 
+4. Click on +App and configure it as follows:
+- Name: Hello World App
+- Developer: Jane Doe (this user was also created during the identity facade deployment)
+- Productt:
+     - Hello World
+     - Identity Facade
+
+
+![Add Product](assets/apigee-app.png)
+
+5. Click Create
+6. Copy the values for Key (client ID) and Secret
+
+![App Keys](assets/app-keys.png)
+
+7. Update the environment variables
+```
```

Comment:
we'll have to fence this code block (to make sure the httpbin.org links aren't validated)

<!-- markdown-link-check-disable -->
```sh
...
```
<!-- markdown-link-check-enable -->